controllers/register_test.go

```diff
@@ -0,0 +1,131 @@
+package controllers
+
+import (
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"strings"
+	"testing"
+
+	"github.com/gin-gonic/gin"
+	"github.com/gutakk/go-google-scraper/models"
+	"github.com/gutakk/go-google-scraper/tests"
+	"github.com/stretchr/testify/suite"
+	"gopkg.in/go-playground/assert.v1"
+	"gorm.io/driver/postgres"
+	"gorm.io/gorm"
+)
+
+func TestDisplayRegister(t *testing.T) {
+	engine := tests.GetRouter(true)
+	new(RegisterController).applyRoutes(engine)
+
+	response := tests.PerformRequest(engine, "GET", "/register", nil, nil)
+	p, err := ioutil.ReadAll(response.Body)
+	pageOK := err == nil && strings.Index(string(p), "<title>Register</title>") > 0
+
+	assert.Equal(t, http.StatusOK, response.Code)
+	assert.Equal(t, true, pageOK)
+}
+
+type DBTestSuite struct {
+	suite.Suite
+	DB       *gorm.DB
+	engine   *gin.Engine
+	formData url.Values
+	headers  http.Header
+}
+
+func (s *DBTestSuite) SetupTest() {
+	db, _ := gorm.Open(postgres.Open(tests.ConstructTestDsn()), &gorm.Config{})
+	s.DB = db
+
+	_ = db.AutoMigrate(&models.User{})
+
+	s.engine = tests.GetRouter(true)
+	registerController := &RegisterController{DB: s.DB}
+	registerController.applyRoutes(s.engine)
+
+	s.headers = http.Header{}
+	s.headers.Set("Content-Type", "application/x-www-form-urlencoded")
+
+	s.formData = url.Values{}
+	s.formData.Set("email", "test@hello.com")
+	s.formData.Set("password", "123456")
+	s.formData.Set("confirm-password", "123456")
+}
+
+func (s *DBTestSuite) TearDownTest() {
+	s.DB.Exec("DELETE FROM users")
+}
+
+func TestDBTestSuite(t *testing.T) {
+	suite.Run(t, new(DBTestSuite))
+}
```

Comment:
> separate methods for each test way better than nesting all tests under one method
Could you please explain more about this?

Reply:
Here, I am saying that the structure with standalone methods (as you did):

```go
func (s *DBTestSuite) TestRegisterWithValidParameters() {}

func (s *DBTestSuite) TestRegisterWithBlankEmail() {}
```

Is better than:

```go
func Test(t *testing.T) {
  t.Run("RegisterWithValidParameters", func(t *testing.T) {

  }

  t.Run("RegisterWithBlankEmai", func(t *testing.T) {

  }
}
```

So that's a good way to move forward with this.